### DISCUSSION TITLE
How to pass value from one step to another?

### DISCUSSION THREAD

**Discussion by bansal63** [getgauge/gauge#2049] [General]
Hi,

I have a column in csv which has values like P1, P2, P3. And based on these values we fetch a id from DB. So P1 gets ID1, P2 gets ID and so on.

I can write a step to get these value from DB, but how can i pass this value ID1 to the next steps?
I do not want to store it in Data store. Is there any other way?

I can think of updating the csv. But i believe thats a overhead if i have to do this for all the csvs.

**Comment by sriv** [comment]
datastore is the only way that gauge can facilitate passing of values. Other than that you may have to use either one of the language features or put in a mechanism using other libraries/design patterns.
